### PR TITLE
Implement plugin activation message

### LINKED ISSUE #2853: Plugin activation notice
## Feature Description

As part of the **first time user experience** (FTUE), we want to add a "getting started" message right after plugin activation to guide the user towards the example story (se #2845).

## Additional Context

Here's how Site Kit does it, for context:

Screenshot 2020-06-30 at 17 18 38

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief

* The notice should be a small React app for consistency reasons.  
  This way we can use styled components and in the future also do the telemetry (#2372) opt-in right away.

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Implements the plugin activation message according to the [designs](https://www.figma.com/file/HEVtT9QucEpRGvzUkCCvL6/Beta-Micro-Site?node-id=101%3A9) shared at #2853

## Relevant Technical Choices

* **For testing & review purposes only:** This PR currently includes a small theme switcher. **Double click on the message** to switch between light and dark theme.  
  This way we can get a better sense for which works best. Also, theoretically we can keep both themes and do A/B testing in the future.
  **This switcher will be removed before merge.**
* The original design does not work on a 15 inch screen. I took some liberty to make some assets **smaller**, and to add some **responsive design** to the whole thing.
* Ensures that all clicks to links within the message can be tracked.

## To-do

* [ ] Implement Get Started story (#2845)
* [ ] Remove theme switcher once settled for a theme.
* [ ] Visual fine tuning

## User-facing changes

Implements a new plugin activation message, check screenshots below:

<details>
<summary>Small Screen (Smartphone)</summary>

Screenshot 2020-07-15 at 11 14 33
Screenshot 2020-07-15 at 11 14 25

</details>

<details>

<summary>Small Tablet</summary>

Screenshot 2020-07-15 at 11 15 04
Screenshot 2020-07-15 at 11 15 14

</details>

<details>
<summary>Large Tablet</summary>

Screenshot 2020-07-15 at 11 15 42
Screenshot 2020-07-15 at 11 15 34

</details>

<details>
<summary>Desktop</summary>

Screenshot 2020-07-15 at 11 15 56
Screenshot 2020-07-15 at 11 16 05

</details>

## Testing Instructions

1. Deactivate plugin.
1. Activate plugin.
1. Verify that all links work (except for Get Started story!)
1. Verify that images are clickable too
1. Verify that plugin activation message is responsive
1. On smaller screens, a "See sample story" link is displayed below the "Launch the editor"  button

---

Fixes #2853

### CODE REVIEW

File: assets/src/activation-notice/app/components/dismiss.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * Renders a Dismiss button as required/used by WordPress.
+ *
+ * Does not actually *render* the button itself, but emits an event so that
+ * WordPress detects the notice and registers a click handler.
+ *
+ * @return {*} Rendered component.
+ */
+function Dismiss() {
+  useEffect(() => {
+    document.dispatchEvent(new Event('wp-updates-notice-added'));
```
Comment (swissspidy) [inline assets/src/activation-notice/app/components/dismiss.js:32]: Hacky, but works to retain default WordPress behavior, which is to dismiss the notice with a small fade/slide effect.

File: assets/src/activation-notice/app/components/step3.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    },
+    [newStoryURL]
+  );
+
+  return (
+    <Wrapper>
+      <Link href={newStoryURL} onClick={onClick}>
+        <Image
+          src={`${assetsURL}images/plugin-activation/editor.png`}
+          alt={__('Screenshot of the Web Stories Editor', 'web-stories')}
```
Comment (swissspidy) [inline assets/src/activation-notice/app/components/step3.js]: All the screenshots seem rather decorative, so perhaps we should use empty `alt` attribute (`alt=""`) everywhere instead.

File: assets/src/activation-notice/app/components/successMessage.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+const Title = styled.h2`
+  font-family: ${({ theme }) => theme.fonts.title.family};
+  font-size: ${({ theme }) => theme.fonts.title.size};
+  line-height: ${({ theme }) => theme.fonts.title.lineHeight};
+  color: ${({ theme }) => theme.colors.primary};
+  margin: 0 0 10px;
+`;
+
+const PrimaryButton = styled.a`
```
Comment (spacedmonkey) [inline assets/src/activation-notice/app/components/successMessage.js]:
```suggestion
const PrimaryButton = styled.a.attrs({ role: 'button' })`
```

File: assets/src/activation-notice/app/components/link.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  font-family: ${({ theme }) => theme.fonts.body.family};
+  font-size: ${({ theme }) => theme.fonts.body.size};
+  line-height: ${({ theme }) => theme.fonts.body.lineHeight};
+  color: ${({ theme }) => theme.colors.link.fg};
+  text-decoration: none;
+  cursor: pointer;
+
+  &:focus,
+  &:hover {
+    color: ${({ theme }) => theme.colors.link.hover.fg};
```
Comment (spacedmonkey) [inline assets/src/activation-notice/app/components/link.js:32]: This hover is very hard to see the difference.

File: assets/src/activation-notice/app/components/step1.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      <Link href={demoStoryURL} onClick={onClick}>
+        <Image
+          src={`${assetsURL}images/plugin-activation/tips.png`}
+          alt={__('Screenshot of the Get Started Story', 'web-stories')}
+          width={170}
+          height={300}
+        />
+      </Link>
+      <ParagraphWrapper>
+        <Number>{1}</Number>
```
Comment (spacedmonkey) [inline assets/src/activation-notice/app/components/step1.js]: Should these numbers be translatable as well? There are langauges that use different characters for number rights?

Comment (swissspidy) [reply assets/src/activation-notice/app/components/step1.js]: Hmm yeah makes sense. Will add.

File: assets/src/activation-notice/app/components/successMessage.js (hunk 3 above)
Comment (swissspidy) [reply assets/src/activation-notice/app/components/successMessage.js]: Hmm I don't think it should have the button role.

It should just look like a button, but is an actual link, so adding a button role would only make it _less_ accessible.

File: includes/Activation_Notice.php [hunk 6]
```diff
@@ -108,10 +108,33 @@ public function enqueue_assets( $hook_suffix ) {
...
+			)
+		);
+
 		return [
 			'id'     => 'web-stories-plugin-activation-notice',
 			'config' => [
-				'isRTL' => is_rtl(),
+				'isRTL'        => is_rtl(),
+				'assetsURL'    => trailingslashit( WEBSTORIES_ASSETS_URL ),
+				'demoStoryURL' => '#',
```
Comment (spacedmonkey) [inline includes/Activation_Notice.php:136]: Add a todo here to change this to something more responsible.

File: assets/src/activation-notice/app/components/successMessage.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+      <ParagraphWithSpace>
+        {__('Welcome to the Web Stories for WordPress beta.', 'web-stories')}
+      </ParagraphWithSpace>
+      <ParagraphWithSpace>
+        <PrimaryButton href={newStoryURL} onClick={onClickPrimary}>
+          {__('Launch the editor', 'web-stories')}
+        </PrimaryButton>
+      </ParagraphWithSpace>
+      <Paragraph>
+        <SecondaryButton href={demoStoryURL} onClick={onClickSecondary}>
```
Comment (spacedmonkey) [inline assets/src/activation-notice/app/components/successMessage.js]: is this a button or a link. Naming is confusing here.

File: assets/src/activation-notice/app/components/dismiss.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/activation-notice/app/components/dismiss.js:32]: If it works. 👍

File: assets/src/activation-notice/app/components/successMessage.js (hunk 7 above)
Comment (swissspidy) [reply assets/src/activation-notice/app/components/successMessage.js]: Will rename.

File: assets/src/activation-notice/app/components/dismiss.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (spacedmonkey) [inline assets/src/activation-notice/app/components/dismiss.js:1]: Should we ignore this file from test coverage? It is a workaround. If not, can we test it somehow?

Comment (swissspidy) [reply assets/src/activation-notice/app/components/dismiss.js:1]: Ignoring from coverage seems unnecessary. Things to test would be that it renders `null` & that it emits an event.

The actual functionality I think can be tested in an E2E test.

**Comment by spacedmonkey** [comment]
There are some issues with RTL. I am not for if we officially support this, but this feels like a simple fix.
Screenshot 2020-07-16 at 12 50 51

**Comment by spacedmonkey** [comment]
This is looking great, some small nit picks and RTL support. Otherwise 👍

**Comment by swissspidy** [comment]
Thanks a lot @spacedmonkey! I believe I addressed all reported issues now.

**Review by spacedmonkey — APPROVED** [review]
👍

**Pull request merged**